### PR TITLE
[FLINK-22894][table-runtime-blink] Fix rankEnd validation of WindowRank to support top1

### PR DESCRIPTION
**Pull request by beyond1920**
## What is the purpose of the change

The pull request aims to a minor fix on rankEnd validation of WindowRank to support top1 

## Brief change log

  - A minor update in `WindowRankOperatorBuilder` to support top1 

## Verifying this change

  - Add ITCase in WindowRankITCase

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by JingsongLi — APPROVED** [review]
Thanks @beyond1920 Looks good to me.

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/rank/window/WindowRankOperatorBuilder.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ public WindowRankOperatorBuilder windowEndIndex(int windowEndIndex) {
                 rankStart > 0,
                 String.format("Illegal rank start %s, it should be positive!", rankStart));
         checkArgument(
-                rankEnd > 1,
+                rankEnd >= 1,
                 String.format("Illegal rank end %s, it should be bigger than 1!", rankEnd));
```
Comment (alpinegizmo) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/rank/window/WindowRankOperatorBuilder.java:129]:
```suggestion
                String.format("Illegal rank end %s, it should be at least 1!", rankEnd));
```

**Review by alpinegizmo — CHANGES_REQUESTED** [review]
The message for the exception also needs to be adapted.

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/rank/window/WindowRankOperatorBuilder.java (hunk 1 above)
Comment (beyond1920) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/rank/window/WindowRankOperatorBuilder.java]: good catch. I've updated the exception message. Thanks a lot.

**Review by alpinegizmo — APPROVED** [review]
LGTM now. +1.

**Comment by beyond1920** [comment]
@flinkbot run azure

**Comment by JingsongLi** [comment]
Thanks @beyond1920 and @alpinegizmo , merging...

**Pull request merged**